upup/pkg/fi/cloudup/awstasks/securitygrouprule.go

```diff
@@ -146,50 +161,16 @@ func (e *SecurityGroupRule) matches(rule *ec2.IpPermission) bool {
 		return false
 	}
 
-	if e.CIDR != nil {
-		match := false
-		for _, ipRange := range rule.IpRanges {
-			if aws.StringValue(ipRange.CidrIp) == *e.CIDR {
-				match = true
-				break
-			}
-		}
-		if !match {
-			return false
-		}
+	if fi.StringValue(e.CIDR) != fi.StringValue(rule.CidrIpv4) {
+		return false
 	}
 
-	if e.IPv6CIDR != nil {
-		match := false
-		for _, ipv6Range := range rule.Ipv6Ranges {
-			if aws.StringValue(ipv6Range.CidrIpv6) == *e.IPv6CIDR {
-				match = true
-				break
-			}
-		}
-		if !match {
-			return false
-		}
+	if fi.StringValue(e.IPv6CIDR) != fi.StringValue(rule.CidrIpv6) {
+		return false
 	}
 
 	if e.SourceGroup != nil {
-		match := false
-		for _, spec := range rule.UserIdGroupPairs {
-			if e.SourceGroup == nil {
-				continue
-			}
-
-			if e.SourceGroup.ID == nil {
-				klog.Warningf("SourceGroup had nil ID: %v", e.SourceGroup)
-				continue
-			}
-
-			if aws.StringValue(spec.GroupId) == *e.SourceGroup.ID {
-				match = true
-				break
-			}
-		}
-		if !match {
+		if fi.StringValue(e.SourceGroup.ID) != fi.StringValue(rule.ReferencedGroupInfo.GroupId) {
```

Comment:
Can `rule.ReferencedGroupInfo` be nil here?

Reply:
I suppose it can in theory. That would mean there is no CIDRs on the rule (as that would have failed the tests above) and no referencing groups though.
But could perhaps change line 172 to also test for the referenced group.

Reply:
I think if `e.SourceGroup == nil && rule.ReferencedGroupInfo != nil` or vice versa we should return false

Reply:
This should do it, I think.